home-manager/home-manager

```diff
@@ -791,8 +791,8 @@ function buildNews() {
     nix-instantiate \
         --no-build-output --strict \
         --eval '<home-manager/home-manager/build-news.nix>' \
-        --arg newsJsonFile "\"$(escapeForNix "$newsJsonFile")\"" \
-        --arg newsReadIdsFile "\"$(escapeForNix "$readIdsFile")\"" \
+        --argstr newsJsonFile "\"$(escapeForNix "$newsJsonFile")\"" \
+        --argstr newsReadIdsFile "\"$(escapeForNix "$readIdsFile")\"" \
```

Comment:
This should be sufficient:

```suggestion
        --argstr newsReadIdsFile "$readIdsFile" \
```

` --argstr` will take the value as a string without any processing so it does not need to be quoted or escaped. Notice that the error shows extra quotes:

       error: string '"/var/folders/bz/h4njq8yd7td8qk42rt_9f0p00000gn/T/home-manager-build.vIGAkEVEQX/news.json"' doesn't represent an absolute path

Reply:
Actually, https://github.com/nix-community/home-manager/commit/1395379a7a36e40f2a76e7b9936cc52950baa1be already fixed it. I missed it because GitHub search gave me an old revision.